tools/faucet/webserver/web_server.go

```diff
@@ -24,80 +25,38 @@ type requestAddr struct {
 	Address string `json:"address" binding:"required"`
 }
 
-func NewWebServer(faucetServer *faucet.Faucet, bindAddress string, jwtSecret []byte) *WebServer {
+func NewWebServer(faucetServer *faucet.Faucet, bindAddress string, jwtSecret []byte, defaultAmount *big.Int) *WebServer {
 	r := gin.New()
 	gin.SetMode(gin.ReleaseMode)
 
-	// todo move this declaration out of this scope
-	parseFunding := func(c *gin.Context) {
-		tokenReq := c.Params.ByName("token")
-		var token string
-
-		// check the token request type
-		switch tokenReq {
-		case faucet.OBXNativeToken:
-			token = faucet.OBXNativeToken
-		case faucet.WrappedOBX:
-			token = faucet.WrappedOBX
-		case faucet.WrappedEth:
-			token = faucet.WrappedEth
-		case faucet.WrappedUSDC:
-			token = faucet.WrappedUSDC
-		default:
-			errorHandler(c, fmt.Errorf("token not recognized: %s", tokenReq), faucetServer.Logger)
-			return
-		}
-
-		// make sure there's an address
-		var req requestAddr
-		if err := c.Bind(&req); err != nil {
-			errorHandler(c, fmt.Errorf("unable to parse request: %w", err), faucetServer.Logger)
-			return
-		}
-
-		// make sure the address is valid
-		if !common.IsHexAddress(req.Address) {
-			errorHandler(c, fmt.Errorf("unexpected address %s", req.Address), faucetServer.Logger)
-			return
-		}
-
-		amount := int64(100)
+	// authed endpoint
+	r.POST("/auth/fund/:token", jwtTokenChecker(jwtSecret, faucetServer.Logger), fundingHandler(faucetServer, defaultAmount))
 
-		// fund the address
-		addr := common.HexToAddress(req.Address)
-		if err := faucetServer.Fund(&addr, token, amount); err != nil {
-			errorHandler(c, fmt.Errorf("unable to fund request %w", err), faucetServer.Logger)
-			return
-		}
+	r.POST("/fund/:token", fundingHandler(faucetServer, defaultAmount))
```

Comment:
Can a comment be added here to take care of this endpoint before deploying to sepolia ?